core/src/core/yaml_loader/parser.py

```
@@ -0,0 +1,88 @@
+            raise FileNotFoundError(msg)
+
+        try:
+            with open(config_path, encoding="utf-8") as file:
+                raw_config = yaml.safe_load(file)
+
+            processed_config = cls._substitute_env_vars(raw_config)
+            pipeline_config = PipelineConfig(**processed_config)
+
+            cls._validate_env_vars(pipeline_config)
+
+            return pipeline_config
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)